operator/src/main/java/oracle/kubernetes/weblogic/domain/model/DomainStatus.java

```diff
@@ -116,28 +117,24 @@ public DomainStatus(DomainStatus that) {
    */
   public DomainStatus addCondition(DomainCondition newCondition) {
     if (conditions.contains(newCondition)) {
-      conditions = conditions.stream()
-          .filter(c -> preserve(c, newCondition.getType().typesToRemoveAlways())).collect(Collectors.toList());
```

Comment:
I am probably misinterpreting, but it looks we can't assume all conditions are preserved in the case where conditions contains the new condition?  The only I reason I say this is that the new 'preserve' function below looks to me like it's careful to remove an old condition from the current set if it is the 'same type' as the new condition and the old condition is not True.

Reply:
Yes; most condition types already replace old versions of themselves in any case, but this ensures that those that don't, such as Failed, will still remove old conditions with status False or Unknown.